source/GameAction.cpp

```diff
@@ -226,6 +226,24 @@ void GameAction::LoadSingle(const DataNode &child)
 		fail.insert(child.Token(1));
 	else if(key == "fail")
 		failCaller = true;
+	else if(key == "attributes")
+	{
+		// Current expected format is attributes add/set <attribute name> <value>
+		if(child.Size() == 4)
+		{
+			if(child.Token(1) == "add")
+			{
+				double valueChange = static_cast<double>(child.Value(3));
+				modifyAttributes[Get(child.Token(2))] = valueChange;
```

Comment:
I think you can do this, though I'm not 100% sure.
```suggestion
				modifyAttributes.Get(child.Token(2)) = valueChange;
```

Reply:
I appreciate the suggestion. And this may end up being needed, although with changes it can't be merged as is. That being said, leaving it here in case I need inspiration and/or reminder of this.